staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -105,67 +104,44 @@ var (
 		},
 		[]string{priorityLevel, flowSchema},
 	)
-	// PriorityLevelExecutionSeatsObserverGenerator creates observers of seats occupied throughout execution for priority levels
-	PriorityLevelExecutionSeatsObserverGenerator = NewSampleAndWaterMarkHistogramsGenerator(clock.RealClock{}, time.Millisecond,
-		&compbasemetrics.HistogramOpts{
+	// PriorityLevelExecutionSeatsObserverVec creates observers of seats occupied throughout execution for priority levels
+	PriorityLevelExecutionSeatsObserverVec = NewTimingRatioHistogramVec(
+		&compbasemetrics.TimingHistogramOpts{
 			Namespace:      namespace,
 			Subsystem:      subsystem,
-			Name:           "priority_level_seat_count_samples",
-			Help:           "Periodic observations of number of seats occupied for any stage of execution (but only initial stage for WATCHes)",
-			Buckets:        []float64{0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1},
+			Name:           "priority_level_seat_utilization",
+			Help:           "Observations, at the end of every nanosecond, of utilization of seats for any stage of execution (but only initial stage for WATCHes)",
+			Buckets:        []float64{0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.99, 1},
```

Comment:
The reason to have both 0.99 and 1 is so that PromQL's `histogram_quantile` (which does not use the `+inf` bucket) will reveal saturation.

I will add the 95% boundary.

I want to keep the deciles for now, since it is still early days for this feature and we need to learn about how it is working out in practice.

Reply:
oh ok, I hadn't thought about that, it might be worth a comment just to make sure that it isn't removed inadvertently in the future.

Yeah now that I understand your reasoning, I think it makes sense to keep the deciles